module/documents/activity/mixin.mjs

```diff
@@ -715,6 +720,29 @@ export default Base => class extends PseudoDocumentMixin(Base) {
     if ( config.consume === false ) return updates;
     const errors = [];
 
+    // Handle action economy
+    if ( ((config.consume === true) || config.consume.action) && (this.activation.type === "legendary") ) {
+      const containsLegendaryConsumption = this.consumption.targets
+        .find(t => (t.type === "attribute") && (t.target === "resources.legact.value"));
+      const count = this.activation.value ?? 1;
+      const legendary = this.actor.system.resources?.legact;
+      if ( legendary && !containsLegendaryConsumption ) {
+        let message;
+        if ( legendary.value === 0 ) message = "DND5E.ACTIVATION.Warning.NoActions";
+        else if ( count > legendary.value ) message = "DND5E.ACTIVATION.Warning.NotEnoughActions";
+        if ( !message ) {
+          updates.actor["system.resources.legact.value"] = legendary.value - count;
+        } else {
+          const err = new ConsumptionError(game.i18n.format(message, {
+            type: game.i18n.localize("DND5E.LegAct"),
+            required: formatNumber(count),
+            available: formatNumber(legendary.value)
+          }));
+          errors.push(err);
+        }
```

Comment:
Prefer if these branches are swapped so that the `else` branch doesn't read like a double-negative.